app/models/shelter.rb

```diff
@@ -16,6 +16,16 @@ def self.order_by_number_of_pets
       .order("pets_count DESC")
   end
 
+  def self.reverse_alpha
+    find_by_sql("select * from shelters order by name desc")
+  end
+
+  def self.only_pending_apps
+    joins(pets: { pet_apps: :application })
+    .where(applications: { status:  1 })
+    .distinct
+  end
```

Comment:
This is a great query, I see that you needed to get to the applications table through application_pets and you made use of your pets association to do that. An easier way to do this would be to add another association to the model:

```erb
has_many :application_pets, through: :pets
has_many :applications, through: :application_pets
```

And with this, you can simplify your query:

```ruby
def self.only_pending_apps
  applications.where(applications: { status: 1 }).distinct
end
```